File: DI/tema8/Select-persona-Azure/app/domain/interfaces/use-cases/IUseCasePersonas.ts
import { Persona } from '../../entities/Persona';
import { PersonaConNombreDeDepartamentoDTO } from '../../dtos/PersonaConNombreDeDepartamentoDTO';

export interface IUseCasePersonas {
  getListaPersonasConDepartamento(): Promise<PersonaConNombreDeDepartamentoDTO[]>;
  getListaPersonas(): Promise<Persona[]>;
  getDetallePersona(id: number): Promise<PersonaConNombreDeDepartamentoDTO | null>;
  crearPersona(persona: Persona): Promise<number>;
  actualizarPersona(id: number, persona: Persona): Promise<number>;
  eliminarPersona(id: number): Promise<number>;
}

File: ListadoPersonasYDepartamentos/Domain/Interfaces/IUseCasePersonas.cs
using Domain.DTOs;
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces
{
    public interface IUseCasePersonas
    {
        List<PersonaConNombreDeDepartamentoDTO> getListaPersonasConDepartamento();

        List<Persona> getListaPersonas();
        PersonaConNombreDeDepartamentoDTO GetDetallePersona(int id);
        PersonaConListaDeDepartamentosDTO GetPersonaConListaDepartamentos(int id);
        PersonaConListaDeDepartamentosDTO GetPersonaParaCrear();

        int CrearPersona(Persona persona);
        int ActualizarPersona(int id, Persona persona);
        int EliminarPersona(int id);
    }
}
